### PR TITLE
Add loader API and registry

### PR DESCRIPTION
**Pull request by tribal-tec**
* Allows plugins to load data from client load RPCs
* Cleanups handling of different data types in LoadDataFunctor
* Extract CircuitLoader from MorphologyLoader
* Removes unused SceneLoader
* Adds support for loading folders from upload-path RPC

### CODE REVIEW

**Comment by tribal-tec** [comment]
retest this please

File: brayns/io/NESTLoader.h [hunk 1]
```diff
@@ -51,11 +51,22 @@ namespace brayns
...
 
+    void importFromBlob(Blob&&, Scene&, const Matrix4f&, const size_t) final
+    {
+        throw std::runtime_error("Unsupported");
+    }
+
+    void importFromFile(const std::string&, Scene&, const Matrix4f&,
+                        const size_t) final
+    {
+        throw std::runtime_error("Unsupported");
```
Comment (rdumusc) [inline brayns/io/NESTLoader.h]: this class also can't load anything? Why is it not calling importCircuit() / importSpikeReport()?

File: brayns/common/loader/LoaderRegistry.cpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        if (!_isSupported(entry, path))
+            continue;
+
+        auto loader = entry.createLoader();
+
+        if (fs::is_directory(path))
+        {
+            fs::directory_iterator begin(path), end;
+            const int numFiles = std::count_if(begin, end, [](const auto& d) {
+                return !fs::is_directory(d.path());
```
[the remark this answers is not in the document]
Comment (tribal-tec) [reply brayns/common/loader/LoaderRegistry.cpp]: Good catch. Pre-filtering will work.

File: brayns/io/NESTLoader.h (hunk 1 above)
Comment (tribal-tec) [reply brayns/io/NESTLoader.h]: exactly, which one to call? The unified load API does not work for some of those 'legacy' loaders. Not even sure if we need that. Hence, for now those loaders are not supported by the 'new' way of loading data. The 'legacy' way of loading is not affected by that, though.

Comment (rdumusc) [reply brayns/io/NESTLoader.h]: But I don't understand what is the blocking factor, is it just lack of time to do it in this PR? If the problem is choosing between 2 methods just make 2 Loader classes, no..?

Comment (tribal-tec) [reply brayns/io/NESTLoader.h]: I'm not spending time on something we don't use atm. I leave it to @favreau to either adapt this loader or remove it.

Comment (favreau) [reply brayns/io/NESTLoader.h]: I would just remove it for now. This will have to go to a plug-in in any case. Don't waste time on that loader.

**Comment by tribal-tec** [comment]
@rdumusc the move of the loading to the loader (doh!) is much nicer in the end. thx for pointing it out!

**Pull request merged**